### ISSUE TITLE
Dropdown Menus could be more accessible

### ISSUE THREAD

**Issue by paulcpederson** [Esri/calcite-web#566]
Just used dropdown menus with Vox and I think there are a couple pretty easy things we could do to make it better:

1. Make the example an actual button, or give it a `role=button` so it doesn't read as "Internal Link"
2. I didn't know this, but there is an aria attribute specifically for this: `aria-haspopup="true"`
3. When the dropdown is open, set `aria-expanded="true"` and when closed use `aria-expanded="false"`
4. Add `role="menu"` to the dropdown nav and `role="menuitem"` to the nav options

There are also some harder things which we could do:

1. fence in tabs when the nav is open so that you can't tab out of the dropdown nav (we do this for modals)
2. Down and up arrows should cycle through options
3. When in focus space, enter, and down arrow should open the menu (we do enter already)

These come from the [menu button guidelines](https://www.w3.org/TR/wai-aria-practices/#menubutton) and the [menu guidelines](https://www.w3.org/TR/wai-aria-practices/#menu). I also found a pretty nice implementation in React here: http://davidtheclark.github.io/react-aria-menubutton/demo/

**Comment by paulcpederson** [comment]
#565 takes care of the easy ones

**Comment by nikolaswise** [comment]
I could see 1) being pretty simple - fencing is rough but we figured it out once.

2) sounds ... tricky. One of those things that I feel should be easy but wont be.
3) Seems totally doable - we have a keyboard bus right now, so we can piggyback on that, extend out what our keypresses do.

**Comment by patrickarlt** [comment]
I did quite a bit of fencing in https://github.com/ArcGIS/calcite-web-ui-kit/blob/master/src/util/dom.js#L226-L249 usage is pretty simple https://github.com/ArcGIS/calcite-web-ui-kit/blob/master/src/Modal/ModalElement.js#L88

**Comment by nikolaswise** [comment]
#570 adds a bunch of the keyboard events we're going to need

**Comment by nikolaswise** [comment]
Closing this issue in favor of one that's just focus-focused.
